
<!DOCTYPE html>

<html class="no-js" lang="en">
<head>
<meta charset="utf-8"/>
<meta content="width=device-width,initial-scale=1" name="viewport"/>
<link href="../img/idec-logo.png" rel="icon"/>
<meta content="mkdocs-1.2.1, mkdocs-material-7.1.10" name="generator"/>
<title>iDEC – Comprehensive Codon Mutagenesis - iDEC Resources Wiki</title>
<link href="../assets/stylesheets/main.1fe995fd.min.css" rel="stylesheet"/>
<link href="../assets/stylesheets/palette.f1a3b89f.min.css" rel="stylesheet"/>
<meta content="#7e56c2" name="theme-color"/>
<link crossorigin="" href="https://fonts.gstatic.com" rel="preconnect"/>
<link href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700%7CRoboto+Mono&amp;display=fallback" rel="stylesheet"/>
<style>:root{--md-text-font-family:"Roboto";--md-code-font-family:"Roboto Mono"}</style>
</head>
<body data-md-color-accent="deep-purple" data-md-color-primary="deep-purple" data-md-color-scheme="" dir="ltr">
<script>function __prefix(e){return new URL("..",location).pathname+"."+e}function __get(e,t=localStorage){return JSON.parse(t.getItem(__prefix(e)))}</script>
<input autocomplete="off" class="md-toggle" data-md-toggle="drawer" id="__drawer" type="checkbox"/>
<input autocomplete="off" class="md-toggle" data-md-toggle="search" id="__search" type="checkbox"/>
<label class="md-overlay" for="__drawer"></label>
<div data-md-component="skip">
<a class="md-skip" href="#idec-comprehensive-codon-mutagenesis">
          Skip to content
        </a>
</div>
<div data-md-component="announce">
</div>
<header class="md-header" data-md-component="header">
<nav aria-label="Header" class="md-header__inner md-grid">
<a aria-label="iDEC Resources Wiki" class="md-header__button md-logo" data-md-component="logo" href="../index.html" title="iDEC Resources Wiki">
<img alt="logo" src="../img/idec-logo-white-bg.png"/>
</a>
<label class="md-header__button md-icon" for="__drawer">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"></path></svg>
</label>
<div class="md-header__title" data-md-component="header-title">
<div class="md-header__ellipsis">
<div class="md-header__topic">
<span class="md-ellipsis">
            iDEC Resources Wiki
          </span>
</div>
<div class="md-header__topic" data-md-component="header-topic">
<span class="md-ellipsis">
            
              iDEC – Comprehensive Codon Mutagenesis
            
          </span>
</div>
</div>
</div>
<label class="md-header__button md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"></path></svg>
</label>
<div class="md-search" data-md-component="search" role="dialog">
<label class="md-search__overlay" for="__search"></label>
<div class="md-search__inner" role="search">
<form class="md-search__form" name="search">
<input aria-label="Search" autocapitalize="off" autocomplete="off" autocorrect="off" class="md-search__input" data-md-component="search-query" data-md-state="active" name="query" placeholder="Search" required="" spellcheck="false" type="text"/>
<label class="md-search__icon md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"></path></svg>
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"></path></svg>
</label>
<button aria-label="Clear" class="md-search__icon md-icon" tabindex="-1" type="reset">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"></path></svg>
</button>
</form>
<div class="md-search__output">
<div class="md-search__scrollwrap" data-md-scrollfix="">
<div class="md-search-result" data-md-component="search-result">
<div class="md-search-result__meta">
            Initializing search
          </div>
<ol class="md-search-result__list"></ol>
</div>
</div>
</div>
</div>
</div>
<div class="md-header__source">
<a class="md-source" data-md-component="source" href="https://github.com/idechq/idec-wiki/" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GitHub
  </div>
</a>
</div>
</nav>
</header>
<div class="md-container" data-md-component="container">
<nav aria-label="Tabs" class="md-tabs" data-md-component="tabs">
<div class="md-tabs__inner md-grid">
<ul class="md-tabs__list">
<li class="md-tabs__item">
<a class="md-tabs__link" href="../index.html">
      Home
    </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="introduction.html">
        Resources
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../team_wiki/wiki_list.html">
        Team Wiki
      </a>
</li>
</ul>
</div>
</nav>
<main class="md-main" data-md-component="main">
<div class="md-main__inner md-grid">
<div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Navigation" class="md-nav md-nav--primary md-nav--lifted" data-md-level="0">
<label class="md-nav__title" for="__drawer">
<a aria-label="iDEC Resources Wiki" class="md-nav__button md-logo" data-md-component="logo" href="../index.html" title="iDEC Resources Wiki">
<img alt="logo" src="../img/idec-logo-white-bg.png"/>
</a>
    iDEC Resources Wiki
  </label>
<div class="md-nav__source">
<a class="md-source" data-md-component="source" href="https://github.com/idechq/idec-wiki/" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GitHub
  </div>
</a>
</div>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../index.html">
        Home
      </a>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle" data-md-toggle="__nav_2" id="__nav_2" type="checkbox"/>
<label class="md-nav__link" for="__nav_2">
<span class="md-nav__icon md-icon"></span>
        
          Resources
        
      </label>
<nav aria-label="Resources" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_2">
<span class="md-nav__icon md-icon"></span>
          
            Resources
          
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="introduction.html">
        Introduction
      </a>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle" data-md-toggle="__nav_2_2" id="__nav_2_2" type="checkbox"/>
<label class="md-nav__link" for="__nav_2_2">
<span class="md-nav__icon md-icon"></span>
<a class="md-nav__link" href="genetic_diversity.html" style="margin: initial; padding: initial; pointer-events: initial">
        
          Creating genetic diversity
        </a>
</label>
<nav aria-label="Creating genetic diversity" class="md-nav" data-md-level="2">
<label class="md-nav__title" for="__nav_2_2">
<span class="md-nav__icon md-icon"></span>
<a class="md-nav__link" href="genetic_diversity.html" style="margin: initial; padding: initial; pointer-events: initial">
          
            Creating genetic diversity
          </a>
</label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="epPCR.html">
        Error-prone PCR
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="gene_shuttling.html">
        Recombinatorial gene and domain shuffling
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle" data-md-toggle="__nav_2_3" id="__nav_2_3" type="checkbox"/>
<label class="md-nav__link" for="__nav_2_3">
<span class="md-nav__icon md-icon"></span>
<a class="md-nav__link" href="lab_selection_methods.html" style="margin: initial; padding: initial; pointer-events: initial">
        
          Laboratory selection methods
        </a>
</label>
<nav aria-label="Laboratory selection methods" class="md-nav" data-md-level="2">
<label class="md-nav__title" for="__nav_2_3">
<span class="md-nav__icon md-icon"></span>
<a class="md-nav__link" href="lab_selection_methods.html" style="margin: initial; padding: initial; pointer-events: initial">
          
            Laboratory selection methods
          </a>
</label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle" data-md-toggle="__nav_2_3_1" id="__nav_2_3_1" type="checkbox"/>
<label class="md-nav__link" for="__nav_2_3_1">
<span class="md-nav__icon md-icon"></span>
<a class="md-nav__link" href="binders.html" style="margin: initial; padding: initial; pointer-events: initial">
        
          Selection of binders
        </a>
</label>
<nav aria-label="Selection of binders" class="md-nav" data-md-level="3">
<label class="md-nav__title" for="__nav_2_3_1">
<span class="md-nav__icon md-icon"></span>
<a class="md-nav__link" href="binders.html" style="margin: initial; padding: initial; pointer-events: initial">
          
            Selection of binders
          </a>
</label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="phage_display.html">
        Phage display
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="FACS_FADS.html">
        Cell surface display
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="ribodisplay.html">
        Ribosome display
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="mRNAdisplay.html">
        cDNA display &amp; mRNA display
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="cisdisplay.html">
        CIS display
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="selex.html">
        SELEX
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle" data-md-toggle="__nav_2_3_2" id="__nav_2_3_2" type="checkbox"/>
<label class="md-nav__link" for="__nav_2_3_2">
<span class="md-nav__icon md-icon"></span>
<a class="md-nav__link" href="enzymes.html" style="margin: initial; padding: initial; pointer-events: initial">
        
          Selection of enzymes and reporters
        </a>
</label>
<nav aria-label="Selection of enzymes and reporters" class="md-nav" data-md-level="3">
<label class="md-nav__title" for="__nav_2_3_2">
<span class="md-nav__icon md-icon"></span>
<a class="md-nav__link" href="enzymes.html" style="margin: initial; padding: initial; pointer-events: initial">
          
            Selection of enzymes and reporters
          </a>
</label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="survival.md">
        By survival
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="biochem.md">
        By biochemical assays
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="CSR_SBL.html">
        By in vitro compartmentalisation
      </a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle" data-md-toggle="__nav_2_4" id="__nav_2_4" type="checkbox"/>
<label class="md-nav__link" for="__nav_2_4">
<span class="md-nav__icon md-icon"></span>
<a class="md-nav__link" href="cont_de_methods.html" style="margin: initial; padding: initial; pointer-events: initial">
        
          Continuous directed evolution methods
        </a>
</label>
<nav aria-label="Continuous directed evolution methods" class="md-nav" data-md-level="2">
<label class="md-nav__title" for="__nav_2_4">
<span class="md-nav__icon md-icon"></span>
<a class="md-nav__link" href="cont_de_methods.html" style="margin: initial; padding: initial; pointer-events: initial">
          
            Continuous directed evolution methods
          </a>
</label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="PACE_PANCE.html">
        PACE
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="insitu_mut.md">
        In situ targeted mutagenesis
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="computational_tools.html">
        Computational tools
      </a>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle" data-md-toggle="__nav_2_6" id="__nav_2_6" type="checkbox"/>
<label class="md-nav__link" for="__nav_2_6">
<span class="md-nav__icon md-icon"></span>
<a class="md-nav__link" href="genome_evolution.html" style="margin: initial; padding: initial; pointer-events: initial">
        
          Genome Evolution
        </a>
</label>
<nav aria-label="Genome Evolution" class="md-nav" data-md-level="2">
<label class="md-nav__title" for="__nav_2_6">
<span class="md-nav__icon md-icon"></span>
<a class="md-nav__link" href="genome_evolution.html" style="margin: initial; padding: initial; pointer-events: initial">
          
            Genome Evolution
          </a>
</label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="mage.md">
        MAGE
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="ALE.md">
        Adaptive Laboratory Evolution
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="scramble.md">
        Scramble
      </a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3" id="__nav_3" type="checkbox"/>
<label class="md-nav__link" for="__nav_3">
<span class="md-nav__icon md-icon"></span>
        
          Team Wiki
        
      </label>
<nav aria-label="Team Wiki" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_3">
<span class="md-nav__icon md-icon"></span>
          
            Team Wiki
          
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../team_wiki/wiki_list.html">
        List of team wikis
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../team_wiki/setup_wiki.html">
        Setting up your team wiki
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../team_wiki/access_repo.html">
        Accessing your team wiki repository
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../team_wiki/gh_pages.html">
        Configuring your GitHub Pages
      </a>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3_5" id="__nav_3_5" type="checkbox"/>
<label class="md-nav__link" for="__nav_3_5">
<span class="md-nav__icon md-icon"></span>
        
          Default wiki tutorial
        
      </label>
<nav aria-label="Default wiki tutorial" class="md-nav" data-md-level="2">
<label class="md-nav__title" for="__nav_3_5">
<span class="md-nav__icon md-icon"></span>
          
            Default wiki tutorial
          
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../team_wiki/mkdocs.html">
        MkDocs &amp; Material for MkDocs
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../team_wiki/fork.html">
        Fork a repository
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../team_wiki/envr.html">
        Environment for editing
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../team_wiki/operations.html">
        MkDocs common operations
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../team_wiki/push.html">
        Stage, commit and push changes
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../team_wiki/pull_request.html">
        Submit a pull request
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../team_wiki/forward.html">
        Moving forward
      </a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#introduction">
    Introduction
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#the-principles-of-comprehensive-codon-mutagenesis">
    The principles of comprehensive codon mutagenesis
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#library-construction">
    Library Construction
  </a>
<nav aria-label="Library Construction" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#pfunkel-4">
    PFunkel 4
  </a>
<nav aria-label="PFunkel 4" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#summary">
    Summary
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#challenges-and-limitations">
    Challenges and limitations
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#exampless">
    Examples(s):
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#inverse-pcr12">
    Inverse PCR12
  </a>
<nav aria-label="Inverse PCR12" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#summary_1">
    Summary
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#challenges-and-limitations_1">
    Challenges and limitations
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#examples">
    Example(s)
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#programmed-allelic-series-pals-mutagenesis6">
    Programmed allelic series (PALS) mutagenesis6
  </a>
<nav aria-label="Programmed allelic series (PALS) mutagenesis6" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#summary_2">
    Summary
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#challenges-and-limitations_2">
    Challenges and limitations
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#exampless_1">
    Examples(s)
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#nicking-mutagenesis5">
    Nicking mutagenesis5
  </a>
<nav aria-label="Nicking mutagenesis5" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#summary_3">
    Summary
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#challenges-and-limitations_3">
    Challenges and limitations
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#exampless_2">
    Examples(s)
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#tiling-primers-for-codon-mutagenesis11">
    Tiling primers for codon mutagenesis11
  </a>
<nav aria-label="Tiling primers for codon mutagenesis11" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#summary_4">
    Summary
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#challenges-and-limitations_4">
    Challenges and limitations
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#exampless_3">
    Examples(s)
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#plasmid-recombineering22">
    Plasmid recombineering22
  </a>
<nav aria-label="Plasmid recombineering22" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#summary_5">
    Summary
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#challenges-and-limitations_5">
    Challenges and limitations
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#exampless_4">
    Examples(s)
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#library-oligo-design-and-synthesis-considerations">
    Library Oligo Design and Synthesis Considerations
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#links-to-protocols-and-research-articles">
    Links to Protocols and Research Articles
  </a>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-content" data-md-component="content">
<article class="md-content__inner md-typeset">
<a class="md-content__button md-icon" href="https://github.com/idechq/idec-wiki/blob/main/docs/resources/CCM.md" title="Edit this page">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25z"></path></svg>
</a>
<h1 id="idec-comprehensive-codon-mutagenesis">iDEC – Comprehensive Codon Mutagenesis</h1>
<h2 id="introduction">Introduction</h2>
<p>Deep mutational scanning (DMS) experiments where large combinatorial libraries are coupled to a functional enrichment and DNA is counted using deep sequencing have enabled the generation of information-rich sequence-function maps allowing researchers to comprehensively study the mutational landscape of proteins<sup id="fnref:1"><a class="footnote-ref" href="#fn:1">1</a></sup>. DMS experiments are not directed evolution in the conventional sense as they typically do not evolutionarily optimize a protein for a particular function through multiple rounds of selection. However, DMS experiments leverage many of the same approaches and yield useful insight into mutational landscapes for guiding subsequent directed evolution experiments. DMS experiments also generate large amounts of data that are highly useful for training machine learning models that can be used to guide directed evolution<sup id="fnref:2"><a class="footnote-ref" href="#fn:2">2</a></sup>.
DMS has been applied to several types of mutational lesions<sup id="fnref:3"><a class="footnote-ref" href="#fn:3">3</a></sup> including codon mutations<sup id="fnref:4"><a class="footnote-ref" href="#fn:4">4</a></sup>,<sup id="fnref:5"><a class="footnote-ref" href="#fn:5">5</a></sup>,<sup id="fnref:6"><a class="footnote-ref" href="#fn:6">6</a></sup> as well as mutations that alter protein topology, including domain insertion<sup id="fnref:7"><a class="footnote-ref" href="#fn:7">7</a></sup>,<sup id="fnref:8"><a class="footnote-ref" href="#fn:8">8</a></sup>, circular permutation<sup id="fnref:9"><a class="footnote-ref" href="#fn:9">9</a></sup>, and truncation<sup id="fnref:10"><a class="footnote-ref" href="#fn:10">10</a></sup>. DMS experiments commonly analyze comprehensive codon mutagenesis (CCM). CCM is the process of generating combinatorial libraries where the amino acid encoded at each residue location within a protein is randomized to all possible amino acids. 
CCM libraries can be generated using a variety of techniques. Nicking mutagenesis, codon tiling PCR11, inverse PCR12, transposon mutagenesis<sup id="fnref2:7"><a class="footnote-ref" href="#fn:7">7</a></sup>,<sup id="fnref2:8"><a class="footnote-ref" href="#fn:8">8</a></sup>,<sup id="fnref2:9"><a class="footnote-ref" href="#fn:9">9</a></sup> and recently CCM libraries have been generated from pools of synthetic oligos that can be generated in high-throughput using oligo library synthesis (OLS) approaches<sup id="fnref2:6"><a class="footnote-ref" href="#fn:6">6</a></sup>,<sup id="fnref3:8"><a class="footnote-ref" href="#fn:8">8</a></sup>. Here we present the basic protocols and tips for designing and constructing comprehensive codon mutagenesis libraries and considerations for enrichment and deep sequencing analysis.</p>
<h2 id="the-principles-of-comprehensive-codon-mutagenesis">The principles of comprehensive codon mutagenesis</h2>
<p>The basic principle of comprehensive codon mutagenesis is to generate a large library of vectors for expressing protein variants with single point mutations. A sufficient number of vectors are generated such that the library contains a protein variant with a mutation at each of the possible residue positions. </p>
<h2 id="library-construction">Library Construction</h2>
<p>There are a variety of approaches for generating comprehensive codon mutagenesis libraries that require different levels of resources. Below is a list with general descriptions of many protocols historically used to generate CCM libraries with links to primary literature and detailed protocols. To help decide which approach best fits your lab’s resources each entry contains a summary of the method, challenges and limitations of the method, and examples where this method was used to generate CCM libraries. </p>
<h3 id="pfunkel-4">PFunkel 4</h3>
<h4 id="summary">Summary</h4>
<p>This is the original comprehensive codon mutagenesis approach.
- PFunkel was inspired by Kunkel site-directed mutagenesis that leverages  uracil-containing, single-stranded DNA (ssDNA) templates to limit the presence of wild-type vectors carried over in transformations<sup id="fnref:13"><a class="footnote-ref" href="#fn:13">13</a></sup> as uracil-containing ssDNA is rapidly degraded in bacteria. PFunkel is a modification to this method that increases mutational efficiency from 50-90%  that is achieved using the Kunkle method to nearly 100% using PFunkel. 
- A uracil-containing ssDNA template is generating by propagating phagemid DNA containing the f1 phage origin and target gene in an <em>Escherichia coli</em> dut-1 ung-1 host and infecting the culture with the M13 helper phage and harvesting ssDNA from isolated phage particles. 
    - <em>E.coli</em> dut-1 ung-1 contains a heat-sensitive dUTPase and lacks uracil DNA glycosylase activity resulting in the production of uracil-containing DNA. 
- Mutagenic oligonucleotide primers containing desired mutations are phosphorylated using T4 Polynucleotide Kinase (PNK) to enable subsequent ligation of DNA products using Taq ligase.
- A mutated second strand is generated by mixing the phosphorylated oligonucleotide and the uracil-containing ssDNA template (4:1 molar ratio) along with PfuTurbo Cx DNA polymerase, Taq ligase, DTT, NAD+, and dNTPs. The mixture is then thermocycled to allow for denaturation, annealing, extension and ligation to generate a mutated second strand with a sealed nick. 
- To generate a complementary mutated strand a second oligonucleotide primer that anneals outside of the mutated gene is then added to the reaction and the denaturation, annealing, extension and ligation cycle is repeated. 
- To remove the uracil-containing ssDNA template uracil DNA glycosylase and exonuclease III are added to the reaction to generate nicks at the uracil containing bases and degrade the ssDNA template. 
- DNA is then directly transformed into chemically competent cells or purified and electroporated into electrocompetent cells.
- Method reportedly ddemonstrate close to 100% level opf efficiency for single codon mutations.
- This method is highly extensible allowing for multiple codon positions to be targeted simultaneously in a single vector by leveraging multiple mutagenic primers and the inability of PfuTurbo to strand-displace at low extension temperatures (&lt;68C). 
Single-tube generation of CCM libraries using pools of mutagenic oligos and using a low oligonucleotide primer to ssDNA template ratio (1:20) ensures only a single codon mutation in each vector. Enables 96-97% of desired mutations to be sampled with no multiple mutations and minimal bias.</p>
<h4 id="challenges-and-limitations">Challenges and limitations</h4>
<ul>
<li>The PFunkel method requires a specialized <em>E.coli</em> strain (<em>E.coli</em> dut-1 ung-1) not typically used in cloning workflows.</li>
<li>Requires specialized methods for isolating DNA from phage particles </li>
<li>Uracil-containing ssDNA yields can vary greatly between templates</li>
<li>Requires a specialized DNA polymerase that can use uracil-containing templates (PfuTurbo Cx HotStart) </li>
<li>Requires additional enzymes not commonly found in cloning workflows (uracil DNA glycosylase)</li>
</ul>
<h4 id="exampless">Examples(s):</h4>
<ul>
<li>CCM of β-lactamase (TEM-1)<sup id="fnref2:4"><a class="footnote-ref" href="#fn:4">4</a></sup></li>
<li>CCM of a 40-codon region of the gene encoding levoglucosan kinase (LGK) from Lipomyces starkeyi<sup id="fnref:14"><a class="footnote-ref" href="#fn:14">14</a></sup></li>
<li>CCM of gene encoding an aliphatic amidase from Pseudomonas aeruginosa (amiE)<sup id="fnref:15"><a class="footnote-ref" href="#fn:15">15</a></sup></li>
</ul>
<h3 id="inverse-pcr12">Inverse PCR<sup id="fnref:12"><a class="footnote-ref" href="#fn:12">12</a></sup></h3>
<h4 id="summary_1">Summary</h4>
<p>A template dsDNA plasmid is mutagenized by an inverse PCR reaction using adjacent, non-overlapping oligonucleotide primers that are complementary to opposite strands of the template and amplify the entire DNA vector. One of the oligonucleotide primers used in the PCR encodes ‘NNK’ (where N is A/C/G/T and K is G/T in equimolar ratio) at a single codon position within the protein and will randomize the encoded amino acid. The template is then phosphorylated and ligated to yield circular vectors that can be moved into cells for functional selection. 
- This method is highly efficient, generating 90-95% of expected mutant vectors with ~5-10% of vectors containing deletions that are the result of primer synthesis errors. 
- This method only requires materials commonly used in cloning workflows</p>
<h4 id="challenges-and-limitations_1">Challenges and limitations</h4>
<ul>
<li>PCR can introduce bias into the library due to bias in annealing primers with mismatches. This challenge can be alleviated by introducing the NNK mutation at the 5’ end of one of the oligonucleotide primers. </li>
<li>Each inverse PCR reaction is able to comprehensively mutate a single codon within a protein, requiring individual PCR reactions for each codon position in a protein to generate a comprehensive library.</li>
<li>Large templates can be challenging to work with without introducing random mutations within the template backbone. Can be partially alleviated by using high-fidelity polymerases or by subcloning libraries into clean vector backbones.   </li>
</ul>
<h4 id="examples">Example(s)</h4>
<ul>
<li>CCM of tumor suppressor phosphatase and tensin homolog (PTEN) and the enzyme thiopurine methyltransferase (TPMT)<sup id="fnref:16"><a class="footnote-ref" href="#fn:16">16</a></sup></li>
</ul>
<h3 id="programmed-allelic-series-pals-mutagenesis6">Programmed allelic series (PALS) mutagenesis<sup id="fnref3:6"><a class="footnote-ref" href="#fn:6">6</a></sup></h3>
<h4 id="summary_2">Summary</h4>
<ul>
<li>Uses microarray-synthesized oligonucleotide pools that ‘tile’ the gene to introduce mutations using mutagenic primer extension. </li>
<li>Uses a uracil-containing template so that template DNA can be selectively degraded. </li>
<li>Oligo pool is designed and commercially synthesized</li>
<li>
<p>A series of 8 steps are needed to produce the full mutagenized library.</p>
<ul>
<li>First, library of mutagenic primers are amplified from the oligo pool using a set of adapter sequences and one adapter is removed using a uracil-containing base and USER enzyme. </li>
<li>Second a uracil-containing ssDNA template of the sense strand is generated by amplifying the wild-type dsDNA template was PCR amplified using an unphosphorylated forward primer and a 5’-phosphorylated reverse primer along with Kapa HiFi U+ HotStart Ready Mix supplemented with dUTPs to incorporate uracil bases. The antisense strand is degraded using lambda exonuclease. </li>
<li>Third the mutagenic oligos are annealed and extended on the wildtype sense template</li>
<li>Fourth the gene fragment is PCR amplified using an upstream primer and uracil containing oligo adapter that can be degraded along with the wildtype template using USER enzyme.</li>
<li>Fifth a sense mutagenesis megaprimer is generated by annealing and extending an outer primer</li>
<li>Sixth a uracil-containing ssDNA template of the antisense strand is generated by amplifying the wild-type dsDNA template was PCR amplified using a 5’-phosphorylated forward primer and an unphosphorylated reverse primer along with Kapa HiFi U+ HotStart Ready Mix supplemented with dUTPs to incorporate uracil bases. The sense strand is degraded using lambda exonuclease. </li>
<li>Seventh the mutagenesis megaprimers are annealed and extended on the uracil-containing wildtype antisense template</li>
<li>Finally in the eighth step the WT template is degraded with USER and the the full length-mutant library is amplified using gene flanking primers.This library is then subcloned into a suitable expression vector.</li>
</ul>
</li>
<li>
<p>For specific details see Supplementary Figure 1 from<sup id="fnref5:6"><a class="footnote-ref" href="#fn:6">6</a></sup>.</p>
</li>
</ul>
<h4 id="challenges-and-limitations_2">Challenges and limitations</h4>
<ul>
<li>Requires specialized enzymes for generating the uracil containing DNA templates (Kapa HiFi U, USER enzyme #M5505)</li>
<li>Oligo pools can be prohibitively expensive, but can be applied to many genes in tandem. </li>
</ul>
<h4 id="exampless_1">Examples(s)</h4>
<ul>
<li>CCM of a 64-codon region of the yeast transcription factor Gal4 gene and CCM of a 393-codon region of the human tumor suppressor p53 gene<sup id="fnref4:6"><a class="footnote-ref" href="#fn:6">6</a></sup></li>
<li>CCM of a 403-codon region of the catalytic site of a hyperthermophilic DNA polymerase from  <em>Thermococcus kodakarensis</em>. Library was used for directed evolution to improve activity on synthesis of α-L-threofuranosyl nucleic acids (TNA) using a modified PALS approach<sup id="fnref:17"><a class="footnote-ref" href="#fn:17">17</a></sup></li>
</ul>
<h3 id="nicking-mutagenesis5">Nicking mutagenesis<sup id="fnref2:5"><a class="footnote-ref" href="#fn:5">5</a></sup></h3>
<h4 id="summary_3">Summary</h4>
<ul>
<li>Overcomes the need for uracil-containing template DNA and some of the specialized enzymes (uracil DNA glycosylase) used by PFunkel and PALS. </li>
<li>Selectively nicks double-stranded DNA template using a pair of strand-specific endonucleases (Nt.BbvCI and Nb.BbvCI) that nick wild type template DNA enabling exonucelase III degradation.</li>
<li>To generate a single-stranded DNA (ssDNA) template a double-stranded DNA (dsDNA) template is treated with Nt.BbvC that selectively nicks a single strand at the BbvCI restriction site and is treated with exonuclease III and exonuclease I to degrade the nicked strand. </li>
<li>Mutagenic oligonucleotide primers containing desired mutations are phosphorylated using T4 Polynucleotide Kinase (PNK) to enable subsequent ligation of DNA products using Taq ligase.</li>
<li>A mutated second strand is generated by mixing the phosphorylated mutagenic oligonucleotide and the ssDNA template (1:20 molar ratio) along with Phusion DNA polymerase, Taq ligase, DTT, NAD+, and dNTPs. The mixture is then thermocycled to allow for denaturation, annealing, extension and ligation to generate a mutated second strand with a sealed nick. </li>
<li>The wild-type ssDNA template is degraded by treating with Nb.BbvCI that selectively nicks the single strand at the BbvCI restriction site and is treated with exonuclease III and exonuclease I to degrade the nicked strand. </li>
<li>A secondary oligonucleotide primer that complements the mutated strand is phosphorylated using T4 Polynucleotide Kinase (PNK) to enable subsequent ligation of DNA products using Taq ligase.</li>
<li>A complementary second strand is generated by mixing the Nb.BbvCI treated mutated stand with the phosphorylated secondary oligonucleotide along with Phusion DNA polymerase, Taq ligase, DTT, NAD+, and dNTPs. The mixture is then thermocycled to allow for denaturation, annealing, extension and ligation to generate a mutated second strand with a sealed nick.</li>
<li>DNA is purified and used is electroporated into electrocompetent cells. </li>
<li>Similar extensibility as that of PFunkel allowing for single, multisite, or single-tube comprehensive codon mutagenesis. </li>
<li>Modified to use microarray-synthesized oligo pools instead of individually synthesized oligos<sup id="fnref2:18"><a class="footnote-ref" href="#fn:18">18</a></sup>,<sup id="fnref2:19"><a class="footnote-ref" href="#fn:19">19</a></sup> </li>
</ul>
<h4 id="challenges-and-limitations_3">Challenges and limitations</h4>
<ul>
<li>Template plasmid must contain a 7-bp BbvCI restriction site to enable nicking strategy.</li>
<li>Requires specialized restriction enzymes not commonly used in cloning workflows  (Nt.BbvCI, Nb.BbvCI)</li>
</ul>
<h4 id="exampless_2">Examples(s)</h4>
<ul>
<li>CCM to two 71-codon regions of the gene encoding an aliphatic amidase from Pseudomonas aeruginosa (amiE) and to an 81-codon region of  β-lactamase (TEM-1) using individually synthesized oligos<sup id="fnref3:5"><a class="footnote-ref" href="#fn:5">5</a></sup></li>
<li>Detailed protocol can be found <a href="https://protocolexchange.researchsquare.com/article/nprot-5125/v1">here</a></li>
<li>CCM of the gene encoding an aliphatic amidase from Pseudomonas aeruginosa (amiE), CCM of a 100-codon region of the anti-Influenza human antibody variable heavy gene UCA9, and targeted mutagenesis of the Arabidopsis thaliana gene encoding abscisic acid receptor PYR1 using microarray-synthesized oligo pools<sup id="fnref:18"><a class="footnote-ref" href="#fn:18">18</a></sup></li>
<li>Detailed protocol can be found <a href="https://en.bio-protocol.org/e3697#biaoti34942">here</a> </li>
<li>CCM of the gene encoding the F capsid and the G spike protein from the bacteriophage ΦX174 using microarray-synthesized oligo pools<sup id="fnref:19"><a class="footnote-ref" href="#fn:19">19</a></sup></li>
</ul>
<h3 id="tiling-primers-for-codon-mutagenesis11">Tiling primers for codon mutagenesis<sup id="fnref:11"><a class="footnote-ref" href="#fn:11">11</a></sup></h3>
<h4 id="summary_4">Summary</h4>
<ul>
<li>This approach involves iterative rounds of low-cycle PCR with pools of mutagenic oligonucleotides that each contain ‘NNN’ triplet at a specific codon site.</li>
<li>Oligos with a randomized ‘NNN’ nucleotide followed by upstream and downstream regions of the gene such that all primers have similar melting temperatures are generated using a python script</li>
<li>Two oligo pools are generated one ‘forward-fragement’ set containing primers binding to the sense strand and one ‘reverse-fragment’ set containing primers that bind to the anti-sense strand. These can be commercially ordered as individual primer and then mixed into equimolar pools or can be ordered as oligo pools from a commercial source.</li>
<li>Two end primers are designed to anneal to the termini of the gene sequence containing BsmBI restriction sites for cloning the library into an expression vector. </li>
<li>The entire gene is amplified using the end primers in a 25 thermocycle reaction and agarose gel purified. </li>
<li>Two fragment PCR reactions are run for using the purified template.
    1) A ‘forward-fragment’ reaction containing the forward mutagenesis primer pool as well as the forward end primer and
    2) a ‘reverse-fragment’ reaction containing the reverse mutagenesis primer pool and the reverse end primer.
    These PCRs are performed for only 7 thermocycles. These PCR products are then diluted 1:4 and used for a ‘joining PCR reaction’ using the two end primers to amplify full length mutant genes. This reaction is performed for 20 thermocycles.</li>
<li>This joining PCR reaction is repeated up to three times, the number of times this process is repeated will control the number of multiple mutations observed within the gene (see the primer design software README for details).  </li>
<li>The product of this reaction is purified and cloned into the expression vector using BsmBI. </li>
<li>This vector pool is then electroporated into ultracompetent cells. </li>
</ul>
<h4 id="challenges-and-limitations_4">Challenges and limitations</h4>
<ul>
<li>This method generates proteins with a <em>Poisson distribution</em> of codon mutations. Some will have a single mutation while some will contain zero or multiple mutations. If the mean mutation per protein is 1 then 37% of the proteins will contain zero mutations, 37% will have one mutation, and the rest will have multiple mutations. </li>
<li>Determining the number of cycles for this process to obtain libraries with desired levels of mutations can require some fine tuning. </li>
</ul>
<h4 id="exampless_3">Examples(s)</h4>
<ul>
<li>CCM of the influenza nucleoprotein<sup id="fnref3:11"><a class="footnote-ref" href="#fn:11">11</a></sup></li>
<li>CCM of the HIV envelope protein<sup id="fnref:20"><a class="footnote-ref" href="#fn:20">20</a></sup></li>
<li>CCM of the receptor binding domain of the SARS-CoV-2 spike glycoprotein<sup id="fnref:21"><a class="footnote-ref" href="#fn:21">21</a></sup></li>
</ul>
<h3 id="plasmid-recombineering22">Plasmid recombineering<sup id="fnref:22"><a class="footnote-ref" href="#fn:22">22</a></sup></h3>
<h4 id="summary_5">Summary</h4>
<ul>
<li>An <em>in vivo</em> mutagenesis approach that is amenable to directed evolution workflows. </li>
<li>Relies on single-strand recombineering to generate mutations. An extensive review on recombineering and MAGE can be found <a href="https://www.zotero.org/google-docs/?0VXLEJ">here</a><sup id="fnref:23"><a class="footnote-ref" href="#fn:23">23</a></sup>.</li>
<li>Mutagenic oligonucleotides are designed to be complementary to the lagging strand of the gene of interest and contain a centrally located ‘NNN’ sequence. </li>
<li>Oligonucleotides are then electroporated into the recombineering ready strain containing the WT plasmid.</li>
<li>Plasmids can then be isolated from the strain or directly enriched for functionality using a selection or screen.</li>
<li>After one round of plasmid recombineering ~29% of reads contained single amino acid mutations with nearly complete coverage of the possible residue mutations. 61% of the sequences corresponded to wild type sequences.</li>
<li>Additional rounds of plasmid recombineering can be used to increase the penetrance of mutations and to generate variants with more than one mutation. </li>
<li>Can also be targeted to specific regions. </li>
</ul>
<h4 id="challenges-and-limitations_5">Challenges and limitations</h4>
<ul>
<li>Requires a recombineering and MAGE ready <a href="https://www.addgene.org/64052/">strain</a> (for example: EcNR2 [Escherichia coli MG1655 mutS::cat (ybhBbioAB)::[cI857 N(cro-ea59)::tetR-bla]]).</li>
<li>Yields libraries that contain positional and mismatch bias that are caused by the mechanism of recombineering relying on oligonucleotides annealing to the locus via homology resulting in nonuniform representation of each mutant. </li>
</ul>
<h4 id="exampless_4">Examples(s)</h4>
<ul>
<li>CCM of iLOV a 110-residue domain of the light, oxygen, voltage (LOV) domains of the <em>A.thaliana</em> phototropin 2 protein<sup id="fnref2:22"><a class="footnote-ref" href="#fn:22">22</a></sup></li>
</ul>
<h3 id="library-oligo-design-and-synthesis-considerations">Library Oligo Design and Synthesis Considerations</h3>
<ul>
<li>CCM libraries require the <em>in silico</em> design of oligos used for constructing the library of protein expression vectors. There are several available tools that facilitate library design.</li>
<li>
<p>Useful software tools for the design of oligo libraries. </p>
<ul>
<li><a href="https://github.com/jbloomlab/CodonTilingPrimers">CodonTilingPrimers</a> - Jesse Bloom Lab - Fred Hutchinson Cancer Research Center<sup id="fnref2:11"><a class="footnote-ref" href="#fn:11">11</a></sup> </li>
<li><a href="https://github.com/schmidt-lab/SPINE">SPINE</a>: Saturated Programmable INsertion Engineering - Daniel Schmidt Lab - U. Minnesota<sup id="fnref4:8"><a class="footnote-ref" href="#fn:8">8</a></sup></li>
<li>Options for deep insertional scan or deep mutational scan MutationMaker - <a href="https://github.com/Merck/Mutation_Maker">Merck 24</a></li>
</ul>
<p>Individually synthesized oligos can be purchased from your typical oligo supplier.  Microarray-synthesized oligo library pools can be commercially synthesized and ordered from companies such as Twist Bioscience, Agilent, IDT, GenScript, and others. </p>
</li>
</ul>
<h2 id="links-to-protocols-and-research-articles">Links to Protocols and Research Articles</h2>
<div class="footnote">
<hr/>
<ol>
<li id="fn:1">
<p>Fowler, D. M. &amp; Fields, S. Deep mutational scanning: a new style of protein science. Nature Methods 11, 801–807 (2014). <a class="footnote-backref" href="#fnref:1" title="Jump back to footnote 1 in the text">↩</a></p>
</li>
<li id="fn:2">
<p>Yang, K. K., Wu, Z. &amp; Arnold, F. H. Machine-learning-guided directed evolution for protein engineering. Nat Methods 16, 687–694 (2019). <a class="footnote-backref" href="#fnref:2" title="Jump back to footnote 2 in the text">↩</a></p>
</li>
<li id="fn:3">
<p>Higgins, S. A. &amp; Savage, D. F. Protein Science by DNA Sequencing: How Advances in Molecular Biology Are Accelerating Biochemistry. Biochemistry 57, 38–46 (2018). <a class="footnote-backref" href="#fnref:3" title="Jump back to footnote 3 in the text">↩</a></p>
</li>
<li id="fn:4">
<p>Firnberg, E. &amp; Ostermeier, M. PFunkel: Efficient, Expansive, User-Defined Mutagenesis. PLoS ONE 7, e52031 (2012). <a class="footnote-backref" href="#fnref:4" title="Jump back to footnote 4 in the text">↩</a><a class="footnote-backref" href="#fnref2:4" title="Jump back to footnote 4 in the text">↩</a></p>
</li>
<li id="fn:5">
<p>Wrenbeck, E. E. et al. Plasmid-based one-pot saturation mutagenesis. Nat Methods 13, 928–930 (2016). <a class="footnote-backref" href="#fnref:5" title="Jump back to footnote 5 in the text">↩</a><a class="footnote-backref" href="#fnref2:5" title="Jump back to footnote 5 in the text">↩</a><a class="footnote-backref" href="#fnref3:5" title="Jump back to footnote 5 in the text">↩</a></p>
</li>
<li id="fn:6">
<p>Kitzman, J. O., Starita, L. M., Lo, R. S., Fields, S. &amp; Shendure, J. Massively parallel single-amino-acid mutagenesis. Nature Methods 12, 203–206 (2015). <a class="footnote-backref" href="#fnref:6" title="Jump back to footnote 6 in the text">↩</a><a class="footnote-backref" href="#fnref2:6" title="Jump back to footnote 6 in the text">↩</a><a class="footnote-backref" href="#fnref3:6" title="Jump back to footnote 6 in the text">↩</a><a class="footnote-backref" href="#fnref4:6" title="Jump back to footnote 6 in the text">↩</a><a class="footnote-backref" href="#fnref5:6" title="Jump back to footnote 6 in the text">↩</a></p>
</li>
<li id="fn:7">
<p>Nadler, D. C., Morgan, S.-A., Flamholz, A., Kortright, K. E. &amp; Savage, D. F. Rapid construction of metabolite biosensors using domain-insertion profiling. Nat Commun 7, 12266 (2016). <a class="footnote-backref" href="#fnref:7" title="Jump back to footnote 7 in the text">↩</a><a class="footnote-backref" href="#fnref2:7" title="Jump back to footnote 7 in the text">↩</a></p>
</li>
<li id="fn:8">
<p>Coyote-Maestas, W., Nedrud, D., Okorafor, S., He, Y. &amp; Schmidt, D. Targeted insertional mutagenesis libraries for deep domain insertion profiling. Nucleic Acids Research 48, 1010–1010 (2020). <a class="footnote-backref" href="#fnref:8" title="Jump back to footnote 8 in the text">↩</a><a class="footnote-backref" href="#fnref2:8" title="Jump back to footnote 8 in the text">↩</a><a class="footnote-backref" href="#fnref3:8" title="Jump back to footnote 8 in the text">↩</a><a class="footnote-backref" href="#fnref4:8" title="Jump back to footnote 8 in the text">↩</a></p>
</li>
<li id="fn:9">
<p>Atkinson, J. T., Jones, A. M., Zhou, Q. &amp; Silberg, J. J. Circular permutation profiling by deep sequencing libraries created using transposon mutagenesis. Nucleic Acids Research 46, e76–e76 (2018). <a class="footnote-backref" href="#fnref:9" title="Jump back to footnote 9 in the text">↩</a><a class="footnote-backref" href="#fnref2:9" title="Jump back to footnote 9 in the text">↩</a></p>
</li>
<li id="fn:10">
<p>Shams, A. et al. Comprehensive deletion landscape of CRISPR-Cas9 identifies minimal RNA-guided DNA-binding modules. http://biorxiv.org/lookup/doi/10.1101/2020.10.19.344077 (2020) doi:10.1101/2020.10.19.344077. <a class="footnote-backref" href="#fnref:10" title="Jump back to footnote 10 in the text">↩</a></p>
</li>
<li id="fn:11">
<p>Bloom, J. D. An Experimentally Determined Evolutionary Model Dramatically Improves Phylogenetic Fit. Molecular Biology and Evolution 31, 1956–1978 (2014). <a class="footnote-backref" href="#fnref:11" title="Jump back to footnote 11 in the text">↩</a><a class="footnote-backref" href="#fnref2:11" title="Jump back to footnote 11 in the text">↩</a><a class="footnote-backref" href="#fnref3:11" title="Jump back to footnote 11 in the text">↩</a></p>
</li>
<li id="fn:12">
<p>Jain, P. C. &amp; Varadarajan, R. A rapid, efficient, and economical inverse polymerase chain reaction-based method for generating a site saturation mutant library. Analytical Biochemistry 449, 90–98 (2014). <a class="footnote-backref" href="#fnref:12" title="Jump back to footnote 12 in the text">↩</a></p>
</li>
<li id="fn:13">
<p>Kunkel, T. A. Rapid and efficient site-specific mutagenesis without phenotypic selection. Proceedings of the National Academy of Sciences 82, 488–492 (1985). <a class="footnote-backref" href="#fnref:13" title="Jump back to footnote 13 in the text">↩</a></p>
</li>
<li id="fn:14">
<p>Kowalsky, C. A. et al. High-Resolution Sequence-Function Mapping of Full-Length Proteins. PLoS ONE 10, e0118193 (2015). <a class="footnote-backref" href="#fnref:14" title="Jump back to footnote 14 in the text">↩</a></p>
</li>
<li id="fn:15">
<p>Wrenbeck, E. E., Azouz, L. R. &amp; Whitehead, T. A. Single-mutation fitness landscapes for an enzyme on multiple substrates reveal specificity is globally encoded. Nat Commun 8, 15695 (2017). <a class="footnote-backref" href="#fnref:15" title="Jump back to footnote 15 in the text">↩</a></p>
</li>
<li id="fn:16">
<p>Matreyek, K. A. et al. Multiplex assessment of protein variant abundance by massively parallel sequencing. Nat Genet 50, 874–882 (2018). <a class="footnote-backref" href="#fnref:16" title="Jump back to footnote 16 in the text">↩</a></p>
</li>
<li id="fn:17">
<p>Nikoomanzar, A., Vallejo, D., Yik, E. J. &amp; Chaput, J. C. Programmed Allelic Mutagenesis of a DNA Polymerase with Single Amino Acid Resolution. ACS Synth. Biol. 9, 1873–1881 (2020). <a class="footnote-backref" href="#fnref:17" title="Jump back to footnote 17 in the text">↩</a></p>
</li>
<li id="fn:18">
<p>Medina-Cucurella, A. V. et al. User-defined single pot mutagenesis using unamplified oligo pools. Protein Engineering, Design and Selection 32, 41–45 (2019). <a class="footnote-backref" href="#fnref:18" title="Jump back to footnote 18 in the text">↩</a><a class="footnote-backref" href="#fnref2:18" title="Jump back to footnote 18 in the text">↩</a></p>
</li>
<li id="fn:19">
<p>Faber, M. S. et al. Saturation Mutagenesis Genome Engineering of Infective ΦX174 Bacteriophage via Unamplified Oligo Pools and Golden Gate Assembly. ACS Synth. Biol. 9, 125–131 (2020). <a class="footnote-backref" href="#fnref:19" title="Jump back to footnote 19 in the text">↩</a><a class="footnote-backref" href="#fnref2:19" title="Jump back to footnote 19 in the text">↩</a></p>
</li>
<li id="fn:20">
<p>Dingens, A. S., Haddox, H. K., Overbaugh, J. &amp; Bloom, J. D. Comprehensive Mapping of HIV-1 Escape from a Broadly Neutralizing Antibody. Cell Host &amp; Microbe 21, 777-787.e4 (2017). <a class="footnote-backref" href="#fnref:20" title="Jump back to footnote 20 in the text">↩</a></p>
</li>
<li id="fn:21">
<p>Starr, T. N. et al. Deep Mutational Scanning of SARS-CoV-2 Receptor Binding Domain Reveals Constraints on Folding and ACE2 Binding. Cell 182, 1295-1310.e20 (2020). <a class="footnote-backref" href="#fnref:21" title="Jump back to footnote 21 in the text">↩</a></p>
</li>
<li id="fn:22">
<p>Higgins, S. A., Ouonkap, S. V. Y. &amp; Savage, D. F. Rapid and Programmable Protein Mutagenesis Using Plasmid Recombineering. ACS Synth. Biol. 6, 1825–1833 (2017). <a class="footnote-backref" href="#fnref:22" title="Jump back to footnote 22 in the text">↩</a><a class="footnote-backref" href="#fnref2:22" title="Jump back to footnote 22 in the text">↩</a></p>
</li>
<li id="fn:23">
<p>Wannier, T. M. et al. Recombineering and MAGE. Nat Rev Methods Primers 1, 7 (2021). <a class="footnote-backref" href="#fnref:23" title="Jump back to footnote 23 in the text">↩</a></p>
</li>
<li id="fn:24">
<p>Hiraga, K. et al. Mutation Maker, An Open Source Oligo Design Platform for Protein Engineering. http://biorxiv.org/lookup/doi/10.1101/2020.06.26.171819 (2020) doi:10.1101/2020.06.26.171819. <a class="footnote-backref" href="#fnref:24" title="Jump back to footnote 24 in the text">↩</a></p>
</li>
</ol>
</div>
</article>
</div>
</div>
</main>
<footer class="md-footer">
<div class="md-footer-meta md-typeset">
<div class="md-footer-meta__inner md-grid">
<div class="md-footer-copyright">
<div class="md-footer-copyright__highlight">
            Copyright © 2021 iDEC. CC-BY-4.0 | The International Directed Evolution Competition (iDEC) is a Scottish Charitable Incorporate Organisation (SCIO), registered in Scotland, with charity number：SC050826
          </div>
        
        Made with
        <a href="https://squidfunk.github.io/mkdocs-material/" rel="noopener" target="_blank">
          Material for MkDocs
        </a>
</div>
</div>
</div>
</footer>
</div>
<div class="md-dialog" data-md-component="dialog">
<div class="md-dialog__inner md-typeset"></div>
</div>
<script id="__config" type="application/json">{"base": "..", "features": ["header.autohide", "navigation.tabs"], "translations": {"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.term.missing": "Missing", "select.version.title": "Select version"}, "search": "../assets/javascripts/workers/search.477d984a.min.js", "version": null}</script>
<script src="../assets/javascripts/bundle.ddd52ceb.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/tablesort/5.2.1/tablesort.min.js"></script>
<script src="../javascripts/tables.js"></script>
<script src="../javascripts/config.js"></script>
<script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
<script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
</body>
</html>